datalad/customremotes/tests/test_archives.py

```diff
@@ -98,6 +99,37 @@ def check_basic_scenario(fn_archive, fn_extracted, direct, d, d2):
     # verify that we can't drop a file if archive key was dropped and online archive was removed or changed size! ;)
 
 
+@with_tree(
+    tree={'a.tar.gz': {'d': {fn_inarchive_obscure: '123'}}}
+)
+def test_annex_get_from_subdir(topdir):
+    from datalad.api import add_archive_content
+    annex = AnnexRepo(topdir, init=True)
+    annex.add('a.tar.gz', commit=True)
+    add_archive_content('a.tar.gz', annex=annex, delete=True)
+
+    with chpwd(opj(topdir, 'a', 'd')):
+        runner = Runner()
+        runner(['git', 'annex', 'drop', fn_inarchive_obscure])
```

Comment:
just in case, could you also please add 'git annex get' call right after here, and verify that the file has needed content (ok_file_has_content  helper), so then we can be sure that both work.